sdks/java/container/boot.go

```diff
@@ -266,9 +267,12 @@ func makePipelineOptionsFile(options string) error {
 // it returns 70% of the physical memory on the machine. If it cannot determine
 // that value, it returns 1GB. This is an imperfect heuristic. It aims to
 // ensure there is memory for non-heap use and other overhead, while also not
-// underutilizing the machine.
-func heapSizeLimit(info *fnpb.ProvisionInfo) uint64 {
-	if size, err := syscallx.PhysicalMemorySize(); err == nil {
+// underutilizing the machine. if set_recommended_max_xmx experiment is enabled, sets xmx 
+// to 32G
+func heapSizeLimit(info *fnpb.ProvisionInfo, setRecommendedMaxXmx bool) uint64 {
+	if setRecommendedMaxXmx {
+		return 32 << 30
```

Comment:
It'll be flexible if we allow passing the xmx value via the experiment.
eg: `--experiments= set_recommended_xmx=<xmx bytes>`

Reply:
We do not want to expose the plug too much with this change as well. Which later becomes hard to control. In future, might be a generic way to control all the jvm args which will be covering this as well.

Reply:
could you please include the description regarding why 32G as a comment here?

Reply:
thanks, I mean code comment

Reply:
Done